internal/channelz/funcs.go

```diff
@@ -155,6 +155,15 @@ func GetSocket(id int64) *SocketMetric {
 	return db.get().GetSocket(id)
 }
 
+// GetServer returns the ServerMetric for the server (identified by id).
```

Comment:
`db.get().GetServers(id, 1)` may not get the result for server with the specified id, it will get the first server with id>= specified id. I think you need to implement a new function `db.get().GetServer(id)` on the `channelMap` struct, which will find the server with the specified id.